File: app/src/main/java/com/aaditx23/bracusocial/backend/local/models/Course.kt
package com.aaditx23.bracusocial.backend.local.models

import io.realm.kotlin.types.RealmObject
import io.realm.kotlin.types.annotations.PrimaryKey
import org.mongodb.kbson.BsonObjectId
import org.mongodb.kbson.ObjectId

class Course: RealmObject{
    @PrimaryKey var _id: ObjectId = BsonObjectId()
    var courseName: String = ""
    var section: String = ""
    var classDay: String? = null
    var classTime: String? = null
    var classRoom: String? = null
    var labDay: String? = null
    var labTime: String? = null
    var labRoom: String? = null
}

File: backend/model/course/course.js
class Course {
    constructor(course, section, faculty, classTime, classRoom, classDay, labTime = "", labRoom = "", labDay = "") {
      this.course = course;
      this.section = section;
      this.faculty = faculty;
      this.classTime = classTime;
      this.classRoom = classRoom;
      this.classDay = classDay;
      this.labTime = labTime;
      this.labRoom = labRoom;
      this.labDay = labDay;
    }
  
    toFirestore() {
      return {
        course: this.course,
        section: this.section,
        faculty: this.faculty,
        classTime: this.classTime,
        classRoom: this.classRoom,
        classDay: this.classDay,
        labTime: this.labTime,
        labRoom: this.labRoom,
        labDay: this.labDay,
      };
    }
  
    static fromFirestore(doc) {
      const data = doc.data();
      return new Course(
        data.course,
        data.section,
        data.faculty,
        data.classTime,
        data.classRoom,
        data.classDay,
        data.labTime || '',
        data.labRoom || '',
        data.labDay || ''
      );
    }
  }
  
  module.exports = Course;
  